subsys/net/lib/lwm2m_client_utils/lwm2m/lwm2m_firmware.c

```diff
@@ -898,18 +916,62 @@ static void lwm2m_firmware_register_write_callbacks(int instance_id)
 {
 	const char *path;
 
-	path = lwm2m_adv_path(instance_id, 0);
+	path = lwm2m_adv_path(instance_id, LWM2M_FOTA_PACKAGE_ID);
 	lwm2m_engine_register_pre_write_callback(path, firmware_get_buf);
-	path = lwm2m_adv_path(instance_id, 1);
+	path = lwm2m_adv_path(instance_id, LWM2M_FOTA_PACKAGE_URI_ID);
 	lwm2m_engine_register_post_write_callback(path, write_dl_uri);
 	/* State */
-	path = lwm2m_adv_path(instance_id, 3);
+	path = lwm2m_adv_path(instance_id, LWM2M_FOTA_STATE_ID);
 	lwm2m_engine_register_post_write_callback(path, firmware_update_state);
 	/* Result */
-	path = lwm2m_adv_path(instance_id, 5);
+	path = lwm2m_adv_path(instance_id, LWM2M_FOTA_UPDATE_RESULT_ID);
 	lwm2m_engine_register_post_write_callback(path, firmware_update_result);
 }
 
+static void firmware_update_check_linked_instances(int instance_id)
+{
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_ADV_FIRMWARE_UPDATE_OBJ_SUPPORT)
+	const char *path;
+	struct lwm2m_objlnk object_link;
+	uint8_t result;
+
+	path = lwm2m_adv_path(instance_id, LWM2M_ADV_FOTA_LINKED_INSTANCES_ID);
+
+	if (lwm2m_engine_get_objlnk(path, &object_link)) {
+		return;
+	}
+
+	if (object_link.obj_id != ENABLED_LWM2M_FIRMWARE_OBJECT) {
```

Comment:
```suggestion
	if (object_link.obj_id != LWM2M_OBJECT_ADV_FIRMWARE_ID) {
```

I know there two things are the same, but it still looks weird, because linked instances only exist in advanced firmware object. Therefore I suggest you use the macro that shows what we are truly comparing against.